src/app/product.service.spec.ts

```diff
@@ -24,23 +26,34 @@ describe('ProductService', () => {
     "totalValue": 1000 
   };
 
-  beforeEach(async () => {
-    await TestBed.configureTestingModule({
+  beforeEach(() => {
+    TestBed.configureTestingModule({
       imports: [
-        HttpClientModule
+        HttpClientTestingModule
+      ],
+      providers: [
+        ProductService
       ]
-    }).compileComponents();
-
-  });
-
-  beforeEach(() => {
-    TestBed.configureTestingModule({});
+    });
+    httpTestingController = TestBed.inject(HttpTestingController);
     service = TestBed.inject(ProductService);
   });
-
+  
   it('should be created', () => {
     expect(service).toBeTruthy();
   });
+  
+  it('should get productsData$ from URL', () => {
+    service.productsData$.subscribe();
```

Comment:
Validate methoda calls inside subscribe

Reply:
What do you mean?

Reply:
service.productsData$.subscribe({
put asserts here :)
});

Reply:
Updated